rpc/cluster/FailsafeTask.java

```diff
@@ -64,7 +65,10 @@ TResult runPrimaryReplica() {
             try {
                 return retries == 0 ? run(replica) : rerun(replica);
             } catch (GraknClientException e) {
-                if (CLUSTER_REPLICA_NOT_PRIMARY.equals(e.getErrorMessage()) || UNABLE_TO_CONNECT.equals(e.getErrorMessage())) {
+                if (CLUSTER_REPLICA_NOT_PRIMARY.equals(e.getErrorMessage()) ||
+                        UNABLE_TO_CONNECT.equals(e.getErrorMessage()) ||
+                        e.getCause() instanceof StatusRuntimeException && ((StatusRuntimeException) e.getCause()).getStatus() == Status.UNKNOWN
+                ) {
```

Comment:
If gRPC has the good, the bad and the ugly parts, this is definitely a cross between bad and ugly.

However, once we introduce our own `Error` message into protocol, we'll be able to replace that logic with just `if e instanceof StatusRuntimeException`.

So let's go ahead with this for now!

Reply:
I've extracted the condition into a function `nonFatalException`